<a href="https://colab.research.google.com/github/othniel77/datascience/blob/main/jointure_de_plusieurs_tables.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import pandas as pd

In [17]:
from google.colab import files
data_to_load = files.upload()

Saving ward.p to ward.p


In [6]:
ridership = pd.read_pickle('/content/cta_ridership.p')
ridership.head()

station_id  year  month  day  rides
0      40010  2019      1    1    576
1      40010  2019      1    2   1457
2      40010  2019      1    3   1543
3      40010  2019      1    4   1621
4      40010  2019      1    5    719

In [7]:
cal = pd.read_pickle('/content/cta_calendar.p')
cal.head()

year  month  day        day_type
0  2019      1    1  Sunday/Holiday
1  2019      1    2         Weekday
2  2019      1    3         Weekday
3  2019      1    4         Weekday
4  2019      1    5        Saturday

In [11]:
stations = pd.read_pickle('/content/stations.p')
stations.head()

station_id        station_name                 location
0      40010  Austin-Forest Park  (41.870851, -87.776812)
1      40020         Harlem-Lake  (41.886848, -87.803176)
2      40030        Pulaski-Lake  (41.885412, -87.725404)
3      40040        Quincy/Wells   (41.878723, -87.63374)
4      40050               Davis   (42.04771, -87.683543)

In [8]:
ridership.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3285 entries, 0 to 3284
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   station_id  3285 non-null   object
 1   year        3285 non-null   int64 
 2   month       3285 non-null   int64 
 3   day         3285 non-null   int64 
 4   rides       3285 non-null   int64 
dtypes: int64(4), object(1)
memory usage: 128.4+ KB


In [9]:
cal.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 365 entries, 0 to 364
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   year      365 non-null    int64 
 1   month     365 non-null    int64 
 2   day       365 non-null    int64 
 3   day_type  365 non-null    object
dtypes: int64(3), object(1)
memory usage: 11.5+ KB


In [12]:
stations.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 144 entries, 0 to 143
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   station_id    144 non-null    object
 1   station_name  144 non-null    object
 2   location      144 non-null    object
dtypes: object(3)
memory usage: 4.5+ KB


In [10]:
# Merge the ridership and cal tables
ridership_cal = ridership.merge(cal)

In [13]:
# Merge the ridership, cal, and stations tables
ridership_cal_stations = ridership.merge(cal, on=['year','month','day']).merge(stations)

Create a variable called filter_criteria to select the appropriate rows from the merged table so that you can sum the rides column.

In [14]:
# Create a filter to filter ridership_cal_stations
filter_criteria = ((ridership_cal_stations['month'] == 7)
                   & (ridership_cal_stations['day_type'] == "Weekday")
                   & (ridership_cal_stations['station_name'] == "Wilson"))

# Use .loc and the filter to select for rides
print(ridership_cal_stations.loc[filter_criteria, 'rides'].sum())

140005


You merged three DataFrames together, including merging two tables on multiple columns. Once the tables were merged, you filtered and selected just like any other DataFrame. Finally, you found out that the Wilson station had 140,005 riders during weekdays in July.

# joindre trois table